tket/src/Predicates/Predicates.cpp

```diff
@@ -525,6 +526,47 @@ std::string MaxNQubitsPredicate::to_string() const {
   return auto_name(*this) + "(" + std::to_string(n_qubits_) + ")";
 }
 
+bool MaxNClRegPredicate::verify(const Circuit& circ) const {
+  bit_vector_t all_bits = circ.all_bits();
+  std::map<std::string, unsigned> bits_map;
```

Comment:
This doesn't need to be a map, does it? Can just be a set of string.

Reply:
I have replaced that with a set and removed the else case as well. Done in @cqc-melf
[requested changes](https://github.com/CQCL/tket/pull/822/commits/14fa39fb728ff26eb74db4d1c34d5eb86cf21c08)
[14fa39f](https://github.com/CQCL/tket/pull/822/commits/14fa39fb728ff26eb74db4d1c34d5eb86cf21c08) and @cqc-melf
[bit sets rename](https://github.com/CQCL/tket/pull/822/commits/7beba097857da497b3833f4f9eea03e8fd1a17b6)
[7beba09](https://github.com/CQCL/tket/pull/822/commits/7beba097857da497b3833f4f9eea03e8fd1a17b6)